src/widgets-service.js

```diff
@@ -150,6 +151,16 @@ WidgetsService.prototype.createTimeSeriesModel = function (attrs, layer) {
   return widgetModel;
 };
 
+WidgetsService.prototype.setWidgetsState = function (state) {
```

Comment:
I agree that the URL logic would probably better be elsewhere, but it only takes the state from it if there isn't a `state` argument passed to the function. I suppose an ajax request would use `setWidgetsState(state_object_from_ajax_req)`?